cyclonedx/model/definition.py

```diff
@@ -15,21 +15,391 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) OWASP Foundation. All Rights Reserved.
 
+import re
 from typing import TYPE_CHECKING, Any, Iterable, Optional, Union
 
 import serializable
 from sortedcontainers import SortedSet
 
 from .._internal.bom_ref import bom_ref_from_str
 from .._internal.compare import ComparableTuple as _ComparableTuple
+from ..exception.model import CycloneDxModelException
+from ..exception.serialization import CycloneDxDeserializationException, SerializationOfUnexpectedValueException
 from ..serialization import BomRefHelper
-from . import ExternalReference
+from . import ExternalReference, Property
 from .bom_ref import BomRef
 
 if TYPE_CHECKING:  # pragma: no cover
     pass
 
 
+class InvalidCreIdException(CycloneDxModelException):
```

Comment:
exceptions should be defined in https://github.com/CycloneDX/cyclonedx-python-lib/tree/main/cyclonedx/exception